internal/gatewayapi/sort.go

```diff
@@ -50,6 +50,8 @@ func sortXdsIRMap(xdsIR XdsIRMap) {
 			// descending order
 			sort.Sort(sort.Reverse(XdsIRRoutes(http.Routes)))
 		}
+
+		sort.SliceStable(ir.TCP, func(i, j int) bool { return ir.TCP[i].Name < ir.TCP[j].Name })
```

Comment:
I think https://github.com/envoyproxy/gateway/pull/535 should handle order and rm the need for this